File: VB/Exercises/Exercise2/AntaField.vb
Namespace Exercises.Exercise2
  Public Class AntaField(Of T)
    Public Property Value As T

    ReadOnly Property SqlValue As String
      Get
        Dim stringValue = TryCast(Value, String)

        If String.IsNullOrEmpty(stringValue) Then
          Return stringValue
        End If

        Return stringValue.Replace("\'", "\'\'")
      End Get
    End Property
  End Class
End Namespace

File: CS/Exercises/Exercise3/AntaField.cs
namespace Exercises.Exercise3;

class AntaField<T>
{
  public T? Value { get; set; }

  public string? SqlValue
  {
    get
    {
      var stringValue = Value as string;

      if(string.IsNullOrEmpty(stringValue))
      {
        return stringValue;
      }

      return stringValue.Replace("\'", "\'\'");
    }
  }
}
